src/test/java/com/wikia/webdriver/testcases/onsitenotifications/OnSiteNotificationsTests.java

```diff
@@ -0,0 +1,206 @@
+    Notifications notificationsList = getNotificationsDesktop();
+
+    Assertion.assertTrue(notificationsList.contains(notification));
+  }
+
+  @Execute(asUser = User.USER_11, onWikia = WIKI_DESKTOP)
+  @Test(groups = DESKTOP)
+  @InBrowser(emulator = Emulator.DESKTOP_BREAKPOINT_BIG)
+  public void userOnDesktopMarksAllNotificationsAsRead() {
+    Notifications notificationsList = getNotificationsDesktop();
+    notificationsList.markAllAsRead();
+    Assertion.assertFalse(notificationsList.isAnyNotificationUnread());
+  }
+
+  /**
+   * Test methods - MOBILE
+   */
+
+  @Execute(asUser =  User.USER_12, onWikia = WIKI_MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = MOBILE)
+  public void userOnMobileReceivesPostReplyNotification() {
+    Notification notification = createPostReplyNotification(User.USER_12, User.USER_2);
+
+    Assertion.assertTrue(getNotificationsMobile().contains(notification));
+  }
+
+  @Execute(asUser = User.USER_12, onWikia = WIKI_MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = MOBILE)
+  public void userOnMobileReceivesPostUpvoteNotification() {
+    Notification notification = createPostUpvoteNotification(User.USER_12, User.USER_2);
+
+    Assertion.assertTrue(getNotificationsMobile().contains(notification));
+  }
+
+  @Execute(asUser = User.USER_12, onWikia = WIKI_MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = MOBILE)
+  public void userOnMobileReceivesReplyUpvoteNotification() {
+    Notification notification = createReplyUpvoteNotification(User.USER_12, User.USER_12, User.USER_2);
+
+    Assertion.assertTrue(getNotificationsMobile().contains(notification));
+  }
+
+  @Execute(asUser = User.USER_12, onWikia = WIKI_MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = MOBILE)
+  public void userOnMobileSeesConsolidatedReplyNotification() {
+    Notification notification = createConsolidatedPostReplyNotification(User.USER_12);
+    Notifications notificationsList = getNotificationsMobile();
+
+    Assertion.assertTrue(notificationsList.contains(notification));
+  }
+
+  /**
+   * Helper methods
+   */
+
+  private PostEntity.Data createPostAs(User user) {
+    return DiscussionsOperations.using(user, driver).createPostWithUniqueData(siteId);
+  }
+
+  private ReplyEntity.Data createReplyToPostAs(PostEntity.Data post, User user) {
+    return DiscussionsOperations.using(user, driver).createReplyToPost(siteId, post);
+  }
+
+  private void upvotePostAs(PostEntity.Data post, User user) {
+    DiscussionsOperations.using(user, driver).upvotePost(siteId, post);
+  }
+
+  private void upvoteReplyAs(ReplyEntity.Data reply, User user) {
+    DiscussionsOperations.using(user, driver).upvoteReply(siteId, reply);
+  }
+
+  private Notification createPostReplyNotification(User postAuthor, User replyAuthor) {
```

Comment:
This is a bit misleading. You're not creating a reply notification but creating a reply and returning expected notification.